src/server/typingsInstaller/typingsInstaller.ts

```diff
@@ -332,6 +338,9 @@ namespace ts.server.typingsInstaller {
                     if (this.log.isEnabled()) {
                         this.log.writeLine(`Installed typings ${JSON.stringify(scopedTypings)}`);
                     }
+
+                    this.writePackageVersion(cachePath);

```

Comment:
the types registry can have a map of ts server version to package version, the package.json tells you what version is already installed.

technically you only need to install if the one on disk maps to a higher one than the version of the server. since they are back ward compatible.